Userland/Libraries/LibGfx/Painter.h

```diff
@@ -161,6 +161,23 @@ class Painter {
 
     int scale() const { return state().scale; }
 
+    template<typename TGetPixelCallback>
+    void fill_pixels(Gfx::IntRect const& region, bool blend, TGetPixelCallback callback)
+    {
+        VERIFY(scale() == 1);
```

Comment:
What guarantees this? If you add a call to fill_pixels() in Userland/Demos/LibGfxScaleDemo, does that work?

Reply:
This is done in painting functions that don't support scaling (which is quite a few).
This one is ambiguous if called on a scaled painter.
Should it scale the rectangle and paint more pixels? That might break the calling function. 
Should it scale the pixels to rectangles? That's more work and is not really scaling.

Reply:
(This does not clash with CSS pixels, which scales everything before painting)